api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/Conv1D.kt

```diff
@@ -116,18 +101,67 @@ public class Conv1D(
 
         val cols = convOutputLength(
             colsCount,
-            kernelSize,
-            paddingInternal,
+            kernelLength,
+            padding,
             strides[1],
             dilations[1]
         )
 
-        return Shape.make(batchSize, cols, filtersInternal.toLong())
+        return Shape.make(batchSize, cols, filters.toLong())
     }
 
     override fun toString(): String =
         "Conv1D(filters=$filters, kernelSize=$kernelSize, strides=${strides.contentToString()}, " +
                 "dilation=${dilations.contentToString()}, activation=$activation, kernelInitializer=$kernelInitializer, " +
                 "biasInitializer=$biasInitializer, kernelShape=${kernel.shape}, biasShape=${bias?.shape}, padding=$padding, " +
                 "biasRegularizer=$biasRegularizer, kernelRegularizer=$kernelRegularizer, activityRegularizer=$activityRegularizer)"
+
+    internal companion object {
+        internal const val EXTRA_DIM = 1
+
+        /** Axis of height for which the extra dimension is added (unsqueezed) before actual
+         * convolution operation and the output from actual implementation are squeezed. */
+        private val squeezeAxis = Squeeze.axis(listOf(EXTRA_DIM.toLong()))
+
+        internal fun expandKernel(kernel: IntArray): IntArray {
+            return kernel.withAdded(EXTRA_DIM - 1, 1)
+        }
+
+        internal fun Ops.expandKernel(kernel: Operand<Float>): Operand<Float> {
```

Comment:
Is it Conv1D specific functions, could they be a part of our local helper framework over the TensorFlow Java API

Reply:
These functions are `Conv1D`/`Conv1DTranspose`, I don't think they are needed outside of these operations.